website/pages/docs/advanced-topics/docker-cli-docker-plugins.md

```diff
@@ -0,0 +1,112 @@
+# Running Docker Plugins With the CLI Docker Image
+
+CloudQuery CLI uses the [Docker CLI](https://docs.docker.com/engine/reference/commandline/cli/) and [Engine API](https://docs.docker.com/engine/api/) to run Docker plugins.
+When using the CloudQuery CLI Docker image, Docker plugins don't work out of the box, as the Docker CLI and Engine API are not available in the container by default.
+This guide will show you how to run Docker plugins with the CloudQuery CLI Docker image using Docker Compose.
+
+## Prerequisites
+
+- [Docker installed (with Docker Compose)](https://docs.docker.com/get-docker/)
+- [CloudQuery CLI](https://docs.cloudquery.io/docs/quickstart)
+
+## Setup
+
+1. Run `cloudquery login` to authenticate with the CloudQuery registry.
+2. Create a file named `spec.yml` with the configuration for the Docker plugin. We will use this spec file to pull the Docker plugin image locally from the private CloudQuery Docker registry. The `airtable` and `postgresql` plugins are used as an example.
+
+```yaml filename="spec.yml"
+kind: source
+spec:
+  name: airtable
+  registry: docker
+  path: docker.cloudquery.io/cloudquery/source-airtable:VERSION_SOURCE_AIRTABLE
+  tables: ["*"]
+  destinations: ["postgresql"]
+---
+kind: destination
+spec:
+  name: "postgresql"
+  path: "cloudquery/postgresql"
+  version: "VERSION_DESTINATION_POSTGRESQL"
+```
+3. Run `cloudquery plugin install spec.yml` to pull the Docker plugin image locally.
+
+## Running a Sync
+
+1. [Create a CloudQuery API Key](https://docs.cloudquery.io/docs/deployment/generate-api-key) to be used with the Docker Compose file.
+2. Create a `docker-compose.yml` file with the following content. The file configures the CLI docker image, the Docker plugin image, a PostgreSQL database and a configuration spec that sets up a connection between the CLI and the Docker plugin using gRPC.
+
+```yaml filename="docker-compose.yml"
+version: '3.1'
+services:
+  cli:
+    container_name: cli
+    image: ghcr.io/cloudquery/cloudquery:latest
+    command: ["sync", "/spec.yml", "--log-console", "--log-format", "json"]
+    environment:
+      CLOUDQUERY_API_KEY: ${CLOUDQUERY_API_KEY}
+      # We reference this environment variable in the `spec.yml` config block below
+      # Other plugins will require other environment variables
+      AIRTABLE_ACCESS_TOKEN: ${AIRTABLE_ACCESS_TOKEN}
+    configs:
+      - spec.yml
+    depends_on:
+      airtable:
+        condition: service_healthy
+      postgres:
+        condition: service_healthy
+  airtable:
+    container_name: airtable
+    image: docker.cloudquery.io/cloudquery/source-airtable:v2.1.2
+    # We use `cloudquery login` and `cloudquery plugin install spec.yml` to pull the image locally
+    pull_policy: never
```

Comment:
Explained in https://github.com/cloudquery/cloudquery/pull/17581#discussion_r1559613735, only our CLI can pull from our private registry since we pass the team name